File: Sorting-Algorithms/Insertion_Sort/Java/InsertionSort.java
/*
    Insertion Sort
    Time Complexity
        Best Case Scenario  : O(n)
        Worst Case Scenario : O(n^2)
*/
import java.util.*;
public class InsertionSort
{
    static void insertion_sort(int arr[],int n)
    {
        for(int i=0;i<n;i++)
        {
            int key=arr[i];
            int j=i-1;
            while(j>=0 && arr[j]>key)
            {
                arr[j+1]=arr[j];
                j--;
            }   
            arr[j+1]=key;
        }
    }
    public static void main(String args[])
    {
        int arr[]={1,-10,-30,20,2,11,15};
        int n=arr.length;
        insertion_sort(arr,n);
        System.out.print("Array after insertion sort : ");
        for(int i=0;i<n;i++)
        {
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }
}

File: Sorting-Algorithms/Insertion_Sort/Python/insertion_sort.py
'''
    Insertion Sort
    Time Complexity
        Best Case Scenario  : O(n)
        Worst Case Scenario : O(n^2)
'''
def insertion_sort(arr,n):
    for i in range(0,n):
        key=arr[i]
        j=i-1
        while(j>=0 and arr[j]>key):
            arr[j+1]=arr[j]
            j-=1
        arr[j+1]=key

arr=[1,-10,-30,20,2,11,15]
n=len(arr)
insertion_sort(arr,n)
print("Array after insertion sort : ", end='')
for i in arr:
    print("{} ".format(i),end="")

print()
